frontend/src/static/js/v_summary.js

```diff
@@ -542,6 +544,49 @@ window.vSummary = {
       }
     },
 
+    // update tmp dates for Safari browsers //
+    updateTmpFilterSinceDate(event) {
+      const since = event.target.value;
+      if (dateFormatRegex.test(since) && since >= this.minDate) {
+        this.tmpFilterSinceDate = since;
+        event.currentTarget.style.removeProperty('border-bottom-color');
+      } else {
+        // invalid since date detected
+        event.currentTarget.style.borderBottomColor = 'red';
+      }
+    },
+
+    updateTmpFilterUntilDate(event) {
+      const until = event.target.value;
+      if (dateFormatRegex.test(until) && until <= this.maxDate) {
+        this.tmpFilterUntilDate = until;
+        event.currentTarget.style.removeProperty('border-bottom-color');
+      } else {
+        // invalid until date detected
+        event.currentTarget.style.borderBottomColor = 'red';
+      }
+    },
+
+    validateInput(event) {
+      const key = event.keyCode;
+      const isIntegerKey = (key >= 48 && key <= 57);
+      const isArrowKey = (key >= 37 && key <= 40);
+      const isOtherKey = (key === 8 || key === 13 || key === 46); // backspace, enter, delete keys
+      if (!(isIntegerKey || isOtherKey || isArrowKey)) {
+        event.preventDefault();
+      }
+    },
+
+    formatInputDate(event) {
+      const date = event.target.value;
```

Comment:
Possible to intercept `delete`/`backspace`, then check if last char is `-` and remove last two chars.

This will make the control more coherent when modifying.

Reply:
Has updated based on the comments:

![date](https://user-images.githubusercontent.com/32864116/61845485-03b5ef00-aed6-11e9-9ae8-bcae500f6ddc.gif)